server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -415,12 +427,22 @@ public Version minCompatibleShardNode() {
         return minCompatibleShardNode;
     }
 
+    /**
+     * Search requests with point-in-time don't require wildcard expansion as the associated indices are resolved already.
+     */
+    public boolean requiresWildcardExpansion() {
+        return pointInTimeBuilder() == null;
+    }
+
     /**
      * Sets the indices the search will be executed on.
      */
     @Override
     public SearchRequest indices(String... indices) {
         validateIndices(indices);
+        if (pointInTimeBuilder() != null && indices.length > 0) {
+            throw new IllegalArgumentException("[indices] cannot be used with point in time");
```

Comment:
I would rephrase to sasy "indices cannot be set when a point in time is provided" I wonder if this should rather be an illegal state exception